src/services/service-dogs.js

```diff
@@ -15,3 +15,23 @@ export async function getDogsById(id) {
   const resp = await client.from('dogs').select(`*`).match({ id }).single();
   return checkError(resp);
 }
+
+export async function updateDog(dog) {
+  const resp = await client.from('dogs').update(dog).eq('id', dog.id);
```

Comment:
Something isn't working properly with your update. Spell this one out exactly like you did with createDog